locked-asset/token-unstake/src/fees_accumulation.rs

```diff
@@ -120,21 +107,21 @@ pub trait FeesAccumulationModule:
         }
 
         let fees_mapper = self.fees_from_penalty_unlocking();
-        if fees_mapper.is_empty() {
+        let total_fees = fees_mapper.get();
+        if total_fees == 0u64 {
             last_send_mapper.set(current_epoch);
 
             return;
         }
 
-        let fees_attributes = fees_mapper.get();
         let energy_factory_addr = self.energy_factory_address().get();
-        let fee_tokens: EsdtTokenPayment = self
-            .energy_factory_proxy(energy_factory_addr)
-            .create_merged_locked_token_for_fees(
-                fees_attributes.token_amount,
-                fees_attributes.attributes.unlock_epoch,
-            )
-            .execute_on_dest_context();
+
+        let fee_tokens: EsdtTokenPayment = self.lock_virtual(
+            self.get_locked_token_id(&energy_factory_addr, BASE_TOKEN_ID_STORAGE_KEY),
+            total_fees,
+            self.blockchain().get_sc_address(),
+            energy_factory_addr,
+        );
 
         let fees_collector_addr = self.fees_collector_address().get();
         let _: IgnoreValue = self
```

Comment:
we do not have to deposit these tokens directly into the fess_collector contract. We might simply tell the fees_collector how much LKMEX it needs to distribute for the users. On claimRewards from fees_collector should the actual NFT create happen.

Reply:
Updated.